lib/url.js

```diff
@@ -156,6 +156,11 @@ function urlParse(url, parseQueryString, slashesDenoteHost) {
   return urlObject;
 }
 
+function isIpv6Hostname(hostname) {
+  return hostname.charCodeAt(0) === CHAR_LEFT_SQUARE_BRACKET &&
+    hostname.charCodeAt(hostname.length - 1) === CHAR_RIGHT_SQUARE_BRACKET;
```

Comment:
```suggestion
    StringPrototypeCharCodeAt(hostname, hostname.length - 1) === CHAR_RIGHT_SQUARE_BRACKET;
```

Similarly at other places, prefer to use primordials.

Reply:
Thanks, the code is updated